Code/src/main/java/nl/utwente/group10/ui/components/FunctionBlock.java

```diff
@@ -31,32 +31,54 @@
 	 * @throws IOException
 	 */
 	public static FunctionBlock newInstance(int numberOfArguments){
-		//TODO Prettier resource loading
-		//TODO better factory
-		FunctionBlock functionBlock = null;
+		FunctionBlock functionBlock;
 		try{
 			functionBlock = (FunctionBlock) FXMLLoader.load(Main.class.getResource("/ui/FunctionBlock.fxml"), null, new TactileBuilderFactory());
 			functionBlock.initializeArguments(numberOfArguments);
 		}catch(IOException e){
 			e.printStackTrace();
+			functionBlock = null;
 		}
 		return functionBlock;
 	}
 	
-	//TODO To be implemented method that will parse code into Haskell interface
-	// and returns the result
+	/**
+	 * Method that creates a newInstance of this class along with it's visual representation
+	 * @param the number of arguments this FunctionBlock can hold
+	 * @param the name of this FunctionBlock
+	 * @return a new instance of this class
+	 * @throws IOException
+	 */
+	public static FunctionBlock newInstance(int numberOfArguments, String name) throws IOException{
+		FunctionBlock functionBlock = newInstance(numberOfArguments);
+		functionBlock.setName(name);
+		
+		return functionBlock;
+	}
+	
+	/**
+	 * Executes this FunctionBlock and returns the output as a String
+	 * @return Output of the Function
+	 */
 	public String executeMethod(){		
 		return new String("DEBUG-OUTPUT");
 	}
 	
+	/**
+	 * Nest another Node object within this FunctionBlock
+	 * @param node to nest
+	 */
 	public void nest(Node node){
 		Pane nestSPace = (Pane) this.lookup("#nest_space");
 		((Label) this.lookup("#label_function_name")).setText("Higher order function");
 		nestSPace.getChildren().add(node);
 	}
 	
-	//Private method to initialize the argument fields for this function block.
-	//All arguments will be stored as Strings.
+	/**
+	 * Private method to initialize the argument fields for this function block.
+	 * All arguments will be stored as Strings.
```

Comment:
Strings will contain exactly what is entered though the user interface into the argument field, Since there is no way to actually set them as of yet but the need to store them is clear the initialisation has already been done. Next commit will contain a less obscure javaDoc.